File: 第二章_排序/排序(4)/有序矩阵查找练习题/Finder.java
import java.util.*;
 
public class Finder {
   public boolean findX(int[][] mat, int n, int m, int x) {
        int num=mat[0][m-1];
        int i=0;//行索引
	int j=m-1;//列索引
        while (i<n&&j>=0){
            num=mat[i][j];
            if (x>num){//这个数比当前数大，则这个数在下一行
                i++;
            }
            if (x<num){//否则在前一列
                j--;
            }
            if (x==num){//若相等，则找到这个数
                return true;
            }
        }
        return false;
    }
}

File: 第二章_排序/排序(4)/有序矩阵查找练习题/Finder.cpp
class Finder {
public:
    bool findX(vector<vector<int> > mat, int n, int m, int x) {
        bool result=false;
		int col=m-1;//列索引
		int row=0;//行索引

		while(col>=0&&row<n){

			if(x==mat[row][col]){//若相等，则找到这个数
				result=true;
				break;
			}

			if(x>mat[row][col]){//这个数比当前数大，则这个数在下一行
				row++;
			}else{//否则在前一列
				col--;
			}
		}

		return result;
    }
};